.vitepress/config.js

```diff
@@ -0,0 +1,172 @@
+                    [ 'meta', { name: 'twitter:site_name', content: siteData.title } ],
+                    [ 'meta', { name: 'twitter:site', content: '@ethereum' } ], // TODO: Replace with EIPs Twitter account, if one exists
+                    [ 'meta', { name: 'twitter:description', content: pageData.description } ],
+                    // Dublin Core
+                    [ 'meta', { name: 'DC.title', content: eipTitle } ],
+                    ...authors.map(author => [ 'meta', { name: 'DC.creator', content: author } ]),
+                    [ 'meta', { name: 'DC.date', content: frontmatter.created.replace("-", "/") } ],
+                    frontmatter.finalized ? [ 'meta', { name: 'DC.issued', content: frontmatter.finalized.replace("-", "/") } ] : [],
+                    [ 'meta', { name: 'DC.format', content: 'text/html' } ],
+                    [ 'meta', { name: 'DC.language', content: 'en-US' } ],
+                    [ 'meta', { name: 'DC.publisher', content: siteData.title } ],
+                    [ 'meta', { name: 'DC.rights', content: 'Public Domain' } ],
+                    // Citation
+                    [ 'meta', { name: 'citation_title', content: eipTitle } ],
+                    ...authors.map(author => [ 'meta', { name: 'citation_author', content: author } ]),
+                    [ 'meta', { name: 'citation_online_date', content: frontmatter.created.replace("-", "/") } ],
+                    frontmatter.finalized ? [ 'meta', { name: 'citation_publication_date', content: frontmatter.finalized.replace("-", "/") } ] : [],
+                    [ 'meta', { name: 'citation_technical_report_institution', content: siteData.title } ],
+                    [ 'meta', { name: 'citation_technical_report_number', content: frontmatter.eip } ],
+                    // LD+JSON
+                    [ 'script', { type: 'application/ld+json' }, JSON.stringify({
+                        "@type": "WebSite",
+                        "url": `https://eips.ethereum.org/${pageData.relativePath}`,
+                        "name": eipTitle,
+                        "description": pageData.description,
+                        "@context": "https://schema.org"
+                    })],
+                ].filter(x => x?.length);
+            } else {
+                return [];
+            }
+        } catch (error) {
+            logger.error(error);
+            throw error;
+        }
+    },
+    async transformPageData(pageData) {
+        try { // Custom error handling needed because of the way VitePress handles runtime errors (i.e. it doesn't)
+            logger.info(`Transforming ${pageData.relativePath}`);
+            
+            pageData = { ...pageData };
+            let { frontmatter } = pageData;
+
+            if (frontmatter.eip) {
+                // Try to read from cache
+                try {
+                    let cache = JSON.parse(fs.readFileSync(`./.vitepress/cache/eips/${frontmatter.eip}.json`));
```

Comment:
If you're in an `async` function, why use the `Sync` variant?

Reply:
Two reasons:

1. The async variant isn't actually asynchronous. It uses a callback.
2. It's okay to block the current context, since other async functions automatically take over.
3. It looks a little bit better.

Reply:
You should never block in an async context. I think you can use `fs.promises.readFile` instead.